src/utils/grammars/lexedMichelson.ne

```diff
@@ -0,0 +1,158 @@
+     'DUUUUUP', 'DUUUUUUP', 'DUUUUUUUP', 'DIIP', 'DIIIP', 'DIIIIP', 'DIIIIIP', 'DIIIIIIP', 'DIIIIIIIP', 'REDUCE'],
+    constantData: ['Unit', 'True', 'False', 'None', 'instruction'],
+    singleArgData: ['Left', 'Right', 'Some'],
+    doubleArgData: ['Pair'],
+    data: ['Unit', 'True', 'False', 'Left', 'Right', 'Pair', 'Some', 'None', 'instruction'],
+    parameter: ["parameter", "Parameter"],
+    storage: ["Storage", "storage"],
+    code: ["Code", "code"],
+    elt: "Elt",
+    number: /-?(?:[0-9]|[1-9][0-9]+)(?:\.[0-9]+)?(?:[eE][-+]?[0-9]+)?\b/,
+    word: /[a-z]+/,
+    string: /"(?:\\["\\]|[^\n"\\])*"/
+});
+%}
+
+# Pass your lexer object using the @lexer option:
+@lexer lexer
+
+main -> instruction {% id %} | data {% id %} | type {% id %} | parameter {% id %} | storage {% id %} | code {% id %}
+parameter -> %parameter _ type ";" {% singleArgKeywordToJson %}
+storage -> %storage _ type ";" {% singleArgKeywordToJson %}
+code -> %code _ subInstruction _ semicolons _ {% d => d[2] %}  | %code _ "{};" {% d => "code {}" %}
+
+type -> 
+    %comparableType {% keywordToJson %} 
+  | %constantType {% keywordToJson %} 
+  | %singleArgType _ type {% singleArgKeywordToJson %}
+  | %lparen _ %singleArgType _ type %rparen {% singleArgKeywordWithParenToJson %}
+  | %doubleArgType _ type _ type {% doubleArgKeywordToJson %}
+  | %lparen _ %doubleArgType _ type _ type %rparen {% doubleArgKeywordWithParenToJson %}
+
+subInstruction -> %lbrace _ (instruction _ %semicolon _):+ %rbrace {% instructionSetToJson %} 
+ #| %lbrace _ instruction (_ %semicolon _ instruction):+ _ %rbrace {% id %}
+  | %lbrace _ instruction _ %rbrace {% d => d[2] %}
+  | %lbrace _ %rbrace {% id %}
+instruction ->
+    subInstruction {% id %}
+  | %instruction {% keywordToJson %}
+  | %instruction _ subInstruction {% singleArgKeywordToJson %}
+  | %instruction _ type {% singleArgKeywordToJson %}
+  | %instruction _ data {% singleArgKeywordToJson %}
+  | %instruction _ type _ type _ subInstruction {% tripleArgKeyWordToJson %}
+  | %instruction _ subInstruction _ subInstruction {% doubleArgKeywordToJson %}
+  | %instruction _ type _ type {% doubleArgKeywordToJson %}
+  | %instruction _ type _ data {% doubleArgKeywordToJson %}
+
+data ->
+    %data {% keywordToJson %}
+  | %data _ data {% singleArgKeywordToJson %}
+  | %data _ data _ data {% doubleArgKeywordWithParenToJson %}
+  | subData {% id %}
+  | subElt {% id %}
+  | %number {% keywordToJson %}
+  | %string {% keywordToJson %}
+subData -> 
+    "{" _ (data ";" _):+ "}" {% instructionSetToJson %}
+  | "(" _ (data ";" _):+ ")" {% instructionSetToJson %}
+subElt -> 
+    "{" _ (elt ";" _):+ "}" {% instructionSetToJson %}
+  | "(" _ (elt ";" _):+ ")" {% instructionSetToJson %}
+elt -> %elt _ data _ data {% doubleArgKeywordToJson  %}
+
+_ -> [\s]:*
+semicolons -> null | semicolons ";"
+
+@{%
+
```

Comment:
Please leave a comment for each of these providing an example Michelson snippet corresponding to the code.

Reply:
Done!